### ISSUE TITLE
How to decrease the RAM usage

### ISSUE THREAD

**Issue by mihirp1998** [junyanz/pytorch-CycleGAN-and-pix2pix#385]
I ran the Cycle GAN model on 32gb CPU ram server and the model seems to occupy a RAM of 19gb for batch size 1. But runs very slow 

Can you please suggest me the right hardware required for this model. How much GB ram GPU is sufficient to train it. And also how many epochs or how much loss is sufficient to get an accuracy equivalent to your pre-trained models.

Also, can I tune any of the following configs to decrease the RAM usage? 

Thanks!

the following is my current config.

```
            batch_size: 1                             
                    beta1: 0.5                           
          checkpoints_dir: ./checkpoints                 
           continue_train: False                         
                 dataroot: ./datasets/horse2zebra/       	[default: None]
             dataset_mode: unaligned                     
                direction: AtoB                          
              display_env: main                          
             display_freq: 400                           
               display_id: 0                             	[default: 1]
            display_ncols: 4                             
             display_port: 8097                          
           display_server: http://localhost              
          display_winsize: 256                           
                    epoch: latest                        
              epoch_count: 1                             
                 fineSize: 256                           
                  gpu_ids: 0                             
                init_gain: 0.02                          
                init_type: normal                        
                 input_nc: 3                             
                  isTrain: True                          	[default: None]
                 lambda_A: 10.0                          
                 lambda_B: 10.0                          
          lambda_identity: 0.5                           
                 loadSize: 286                           
                       lr: 0.0002                        
           lr_decay_iters: 50                            
                lr_policy: lambda                        
         max_dataset_size: inf                           
                    model: cycle_gan                     
               n_layers_D: 3                             
                     name: maps_cyclegan                 	[default: experiment_name]
                      ndf: 64                            
                     netD: basic                         
                     netG: resnet_9blocks                
                      ngf: 64                            
                    niter: 100                           
              niter_decay: 100                           
               no_dropout: True                          
                  no_flip: False                         
                  no_html: False                         
                 no_lsgan: False                         
                     norm: instance                      
              num_threads: 4                             
                output_nc: 3                             
                    phase: train                         
                pool_size: 50                            
               print_freq: 100                           
           resize_or_crop: resize_and_crop               
          save_epoch_freq: 5                             
         save_latest_freq: 5000                          
           serial_batches: False                         
                   suffix:                               
         update_html_freq: 1000                          
                  verbose: False      
```

**Comment by junyanz** [comment]
- GPU: We can train the CycleGAN on a GPU (e.g., GTX 1080) with 8GB memory. I haven't tested the CPU training performance. 
- the number of epochs (training details): See this [Q & A](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/qa.md#experiment-details-eg-bw-color-306).
- loss function: see this [Q & A.](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/qa.md#why-does-my-training-loss-not-converge-335-164-30)

**Comment by mihirp1998** [comment]
Thanks!
